src/main/java/seedu/socket/model/project/ProjectDeadline.java

```diff
@@ -39,10 +51,11 @@ private Boolean isValidProjectDeadline(String test) {
      */
     public LocalDateTime toLocalDateTime() {
         if (deadline == "") {
-            return null;
+            return null; // this is dangerous TODO consider rethinking implementation
```

Comment:
Can consider not having this method at all and only parsing the string if and when a LocalDateTime object is required. Alternatively, can use an assertion or throw an exception to ensure this method only returns when the deadline string is not empty.

Reply:
In favor of assertion or exceptions
@cmang12 @huanghao1998 @nus-jeff 
thoughts?

Reply:
Modified to throw DateTimeParseException for now, do feel free to suggest alternative implementations, perhaps when the use of this method is clearer